ggml/src/ggml.c

```diff
@@ -6514,6 +6462,12 @@ static void gguf_free_kv(struct gguf_kv * kv) {
 }
 
 struct gguf_context * gguf_init_empty(void) {
+    if (sizeof(float)  != 4) {
+        GGML_ABORT("support for floats with != 32 bits not implemented");
+    }
+    if (sizeof(double) != 8) {
+        GGML_ABORT("support for doubles with != 64 bits not implemented");
+    }
```

Comment:
Strictly speaking this is not enough, because floats may be represented in a different format even if the size matches. In C there is `__STDC_IEC_559__` to check this, but it seems that not every compiler sets this, so it cannot be used reliably. In C++ it should be possible to use `std::numeric_limits<T>::is_iec559` to check that both `float` and `double` are compliant.

It should probably be a `static_assert` somewhere, too much code assumes ieee-754 floats anyway, it's not just the GGUF loader.